sklearn/compose/tests/test_target.py

```diff
@@ -342,6 +342,8 @@ def test_transform_target_regressor_count_fit(check_inverse):
 
 
 class DummyRegressorWithExtraFitParams(DummyRegressor):
+    _metadata_request__check_input = {"fit": "check_input"}
```

Comment:
If the signature checking works, really we don't need this kind of a declaration on a per-name basis. I think these need to match the specification per-method, i.e. `_fit_requests_config = {"sample_weight": REQUESTED, "check_input": REQUESTED}`.

`check_input` is a very interesting case. Its use cases might need some thinking about...

Reply:
How do you propose we handle overriding a `_fit_requests_config` in inheritance? I was hoping to not have to deal with it when we have an attribute per metadata, we pretty much just ignore whatever's been there in the parent.

Reply:
> check_input is a very interesting case. Its use cases might need some thinking about...

I'd probably move it to be an `__init__` parameter.